docs/operate-and-deploy/installation/upgrading.md

```diff
@@ -42,6 +42,30 @@ upgrading a cluster involves leaving the old cluster running on the old version,
 cluster on the new version, porting across your database schema, and finally thinking about your data.
 Read on for details.
 
+
+## (1) In-Place upgrade
+
+#### Steps :-
+
+1. Download the [new ksqlDB version](https://docs.confluent.io/platform/current/installation/installing_cp/zip-tar.html#get-the-software) to upgrade to.
+2. Set it up with all required configs. In our case we mostly copied all configs from the current version(which is running) to the new version.
+3. Take Backup and then copy all UDF JAR to new version .(OPTIONAL STEP In case you don't have any UDF).
+4. Stop ksqlDB process of current version(5.5 in our case) and start newer version(6 in our case) with latest KsqlDB config file which you created in step2.
+5. KsqlDB will start executing queries and rebuild all states using command topic because the value of "ksql.service.id" is the same.
+6. Now KsqlDB Server with the latest version is up and running.
+
+#### Troubleshooting an Upgrade :-
+
+1. If the ksqlDB server fails to start with an error message such as "KsqlException: UdfFactory not compatible with existing factory. function: <FUNCTION_NAME> existing". 
+This error is because Confluent KsqlDB 6 has these functions as in built so it will create conflict with your UDFs. So to solve this you can create a new udf jar by eliminating those inbuilt functions and deploy it and then try to start KsqlDB again.
```

Comment:
```suggestion
This error is because the new ksqlDB version has introduced an in-built function of the same name, creating a conflict with your UDF. To solve this you can create a new UDF jar by renaming or eliminating the conflicting function(s), and restarting ksqlDB with the new UDF jar.
```
Minor rewrite to remove mention of a specific ksqlDB version, and clarify that the conflict is a name conflict :)